odl/test/trafos/fourier_test.py

```diff
@@ -858,5 +859,98 @@ def fhat(x):
     assert np.allclose(ft_f_n, fhat(recip_n.coord_vectors[0]))
 
 
+# ---- Non-uniform Fourier Transform ---- #
+def test_non_uniform_fourier_trafo_2d_call():
+    im_size = 512
+    shape = [im_size] * 2
+    samples = np.array(
+        np.where(np.random.normal(size=shape) >= 0),
+        dtype=float,
+    ).T
+    samples /= im_size
+    samples -= 0.5
+    image = np.random.normal(size=shape)
+    nfft = NonUniformFourierTransform(shape=shape, non_uniform_samples=samples)
+    nfft(image)
+
+
+def test_non_uniform_fourier_adj_trafo_2d_call():
+    im_size = 512
+    shape = [im_size] * 2
+    samples = np.array(
+        np.where(np.random.normal(size=shape) >= 0),
+        dtype=float,
+    ).T
+    samples /= im_size
+    samples -= 0.5
+    nfft_coeffs = np.random.normal(size=(len(samples),))
+    nfft_adj = NonUniformFourierTransformAdjoint(
+        shape=shape,
+        non_uniform_samples=samples,
+    )
+    nfft_adj(nfft_coeffs)
+
+
+def test_non_uniform_fourier_trafo_1d_call():
+    sig_size = 512
+    shape = [sig_size]
+    samples = np.array(
+        np.where(np.random.normal(size=shape) >= 0),
+        dtype=float,
+    ).T
+    samples /= sig_size
+    samples -= 0.5
+    image = np.random.normal(size=shape)
+    nfft = NonUniformFourierTransform(shape=shape, non_uniform_samples=samples)
+    nfft(image)
+
+
+def test_non_uniform_fourier_adj_trafo_1d_call():
+    sig_size = 512
+    shape = [sig_size]
+    samples = np.array(
+        np.where(np.random.normal(size=shape) >= 0),
+        dtype=float,
+    ).T
+    samples /= sig_size
+    samples -= 0.5
+    nfft_coeffs = np.random.normal(size=(len(samples),))
+    nfft_adj = NonUniformFourierTransformAdjoint(
+        shape=shape,
+        non_uniform_samples=samples,
+    )
+    nfft_adj(nfft_coeffs)
+
+
+def test_non_uniform_fourier_trafo_1d_res():
+    sig_size = 512
+    shape = [sig_size]
+    samples = np.array(np.where(np.ones(shape) >= 0), dtype=float).T
```

Comment:
This is a complicated way of writing `np.arange(sig_size)[:, None].astype(float)`.

Reply:
I liked how it related to the previous tests, but I see how it can be disturbing. I will change it for `test_non_uniform_fourier_trafo_2d_res` as well.